akka-projection-samples/src/main/scala/akka/projection/Samples.scala

```diff
@@ -0,0 +1,168 @@
+package akka.projection
+
+import akka.{ Done, NotUsed }
+import akka.actor.ActorSystem
+import akka.kafka.{ CommitterSettings, ConsumerSettings, Subscriptions }
+import akka.kafka.scaladsl.{ Committer, Consumer }
+import akka.persistence.cassandra.query.scaladsl.CassandraReadJournal
+import akka.persistence.jdbc.query.scaladsl.JdbcReadJournal
+import akka.persistence.query.scaladsl.EventsByTagQuery
+import akka.persistence.query.{ EventEnvelope, Offset, PersistenceQuery, TimeBasedUUID }
+import akka.projection.KafkaAtLeastOnceSample.projection
+import akka.projection.KafkaSourceWithTransactionalJDBCSample.{ projection, system }
+import akka.projection.scaladsl.jdbc.{ JdbcOffsetStore, JdbcOffsetStoreAkkaOffset, JdbcOffsetStoreLong, JdbcProjection }
+import akka.projection.scaladsl.kafka.{ KafkaConsumer, KafkaProjection }
+import akka.stream.scaladsl.Source
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.common.TopicPartition
+
+import scala.concurrent.Future
+
+/**
+ * A Kafka projection that commits the offset in the topic using the Committer.flow (at-least-once)
+ */
+object KafkaAtLeastOnceSample {
+  implicit val system = ActorSystem("sample")
+  implicit val ec = system.dispatcher
+
+  val consumerSettings: ConsumerSettings[String, String] = ???
+  val committerSettings = CommitterSettings(system).withMaxBatch(100)
+
+  val src = KafkaConsumer.atLeastOnce(
+    consumerSettings.withGroupId("group1"),
+    Subscriptions.topics("topic1"),
+    committerSettings) { msg =>
+    Future.successful("pretending that this is projected to somewhere")
+  }
+
+  // a pure Kafka projection is super simple, as it only requires a Source
+  val projection = KafkaProjection(src)
+  projection.start()
+  projection.stop()
+}
```

Comment:
This is the same as in the kaka tests. It's an at-least-once Kafka source that used the Committer.flow. 

In the end, since the offset if managed in Kafka the Source contain all the logic and Akka Projection only offers an API to start and stop it. 

The Source can be of arbitrary complexity. 

`KafkaConsumer.atLeastOnce` is a non-existent API. I added it here just to exemplify a possible improvement in Alpakka Kafka. That could be a good dual for `Consumer.atMostOnce`.